includes/class-sensei-course.php

```diff
@@ -2712,12 +2712,24 @@ public static function single_course_content( $content ) {
 	 * @since 1.9.0
 	 */
 	public static function the_course_lessons_title() {
-
 		if ( ! is_singular( 'course' ) ) {
 			return;
 		}
 
+		/**
+		 * Access check for the course lessons.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param bool $show_lessons Whether or not the lessons should be shown. Default true.
+		 * @param int  $course_id    Course ID.
+		 */
+		if ( ! apply_filters( 'sensei_course_lessons_has_access', true, get_the_ID() ) ) {
```

Comment:
I find the hook name unclear. How about something like `sensei_course_show_lessons_list`?

Reply:
I originally had something like that, but then thought it should be consistent with our [existing filter](https://github.com/Automattic/sensei/blob/d9ac8689910bec5b4465bfe29b7a52756af3ba9d/includes/class-sensei-course.php#L2695) that decides whether to show the course content or an excerpt on the single course page.

Reply:
I decided clarity is more important and renamed the filter to `sensei_course_show_lessons`. We should really document some naming conventions for hooks to enforce some sort of consistency.